src/lifx.d.ts

```diff
@@ -1,367 +1,328 @@
+   * @param effectName sets the desired effect, currently available options are: MOVE, OFF
+   * @param speed sets duration of one cycle of the effect, the higher the value the slower the effect animation
+   * @param direction whether to animate from or towards the controller, available options are: TOWARDS, AWAY
+   * @param callback called when light did receive message
    */
-  setMultiZoneEffect(effectName: 'MOVE' | 'OFF', speed: number, direction: 'TOWARDS' | 'AWAY', callback?: () => void);
+  setMultiZoneEffect(effectName: 'MOVE' | 'OFF', speed: number, direction: 'TOWARDS' | 'AWAY', callback: (err: Error) => void): void;
 
   /**
   * Turns the light off
   * @example light('192.168.2.130').off()
-  * @param {Number} [duration] transition time in milliseconds
-  * @param {Function} [callback] called when light did receive message
+  * @param duration transition time in milliseconds
+  * @param callback called when light did receive message
   */
-  off(duration: any, callback: any): void;
+  off(duration: number, callback: (err: Error) => void): void;
 
   /**
   * Turns the light on
   * @example light('192.168.2.130').on()
-  * @param {Number} [duration] transition time in milliseconds
-  * @param {Function} [callback] called when light did receive message
+  * @param duration transition time in milliseconds
+  * @param callback called when light did receive message
   */
-  on(duration: any, callback: any): void;
+  on(duration: number, callback: (err: Error) => void): void;
 
   /**
   * Sets the label of light
   * @example light.setLabel('Kitchen')
-  * @param {String} label new label to be set, maximum 32 bytes
-  * @param {Function} [callback] called when light did receive message
+  * @param label new label to be set, maximum 32 bytes
+  * @param callback called when light did receive message
   */
-  setLabel(label: any, callback: any): void;
+  setLabel(label: string, callback: (err: Error) => void): void;
 
   /**
   * Reboots the light
-  * @param {Function} callback called when light did receive message
+  * @param callback called when light did receive message
   */
-  reboot(callback: any): void;
+  reboot(callback: (err: Error) => void): void;
+}
+
+export interface Product {
+  vendorName: string;
+  productName: string;
+  productFeatures: ProductFeatures;
+  productUpgrades: ProductUpgrade[];
+}
+
+export interface ProductFeatures {
+  hev: boolean;
+  color: boolean;
+  chain: boolean;
+  matrix: boolean;
+  relays: boolean;
+  buttons: boolean;
+  infrared: boolean;
+  multizone: boolean;
+  temperature_range: [number, number];
+  extended_multizone?: boolean;
+  min_ext_mz_firmware?: number;
+  min_ext_mz_firmware_components?: number[];
 }
 
-export const constants: {
+export interface ProductUpgrade {
+  major: number;
+  minor: number;
+  features: ProductFeatures;
+}
+
+export interface ColorValues {
+  hue: number;
+  saturation: number;
+}
+
+export class constants {
```

Comment:
Revert this back to `const` as that's what it is. We're not instantiating it anywhere. This could also break someones code ( I think).